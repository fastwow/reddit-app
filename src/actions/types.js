export const FETCH_POSTS_STARTED = 'fetch_posts_started';
export const FETCH_POSTS_SUCCESS = 'fetch_posts_success';
export const FETCH_MORE_POSTS_STARTED = 'fetch_more_posts_started';
export const FETCH_MORE_POSTS_SUCCESS = 'fetch_more_posts_success';
export const FETCH_REFRESHED_POSTS_STARTED = 'fetch_refreshed_posts_started';
export const FETCH_REFRESHED_POSTS_SUCCESS = 'fetch_refreshed_posts_success';
export const FETCH_FAVORITES_POSTS_STARTED = 'fetch_favorites_posts_started';
export const FETCH_FAVORITES_POSTS_SUCCESS = 'fetch_favorites_posts_success';
export const ADD_TO_FAVORITE_SUCCESS = 'add_to_favorite_success';
export const REMOVE_FROM_FAVORITE_SUCCESS = 'remove_from_favorite_success';
export const FILTER_POSTS_SUCCESS = 'search_post_success';
