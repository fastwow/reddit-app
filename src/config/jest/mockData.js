export const postsResponse = {
  children: [
    {
      data: {
        url: 'https://v.redd.it/1zhxx58rhp421',
        id: '98vj9e',
        thumbnail: 'https://b.thumbs.redditmedia.com/NFJ5zzCUnxTgAl71EkjSSCePc7wtcCLqTdbYgrwJkdU.jpg',
        title: 'Blinks of affection',
      },
    },
    {
      data: {
        url: 'https://i.imgur.com/GsoJi0d.gifv',
        id: 'a6q94b',
        thumbnail: 'https://b.thumbs.redditmedia.com/cjYmtjKiy9OT1y43RFMuTkcH_WmTM66RsSuN7c6xkhA.jpg',
        title: 'This church looks like a gingerbread house.',
      },
    },
  ],
};

export const posts = postsResponse.children.map(c => c.data);
