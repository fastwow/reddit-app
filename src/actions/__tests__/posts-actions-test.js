import mockStore from 'redux-mock-store';
import {
  fetchPosts,
  fetchPostsActions,
  fetchMorePosts,
  fetchMorePostsActions,
  fetchRefreshedPosts,
  fetchRefreshedPostsActions,
} from '../postsActions';
import {FETCH_POSTS, FETCH_MORE_POSTS, FETCH_REFRESHED_POSTS} from '../types';
import {postsResponse} from '../../config/jest/mockData';

const store = mockStore();
const axios = require('axios');
const MockAdapter = require('axios-mock-adapter');
const mock = new MockAdapter(axios);

beforeEach(() => {
  store.clearActions();
});

it('should handle ' + FETCH_POSTS.success + ' action', async () => {
  mock.onGet().reply(200, postsResponse);
  await store.dispatch(fetchPosts(fetchPostsActions));
  expect(store.getActions()).toMatchSnapshot();
});

it('should handle ' + FETCH_POSTS.failure + ' action', async () => {
  const mockError = new Error('Look ma! I am an error.');
  mock.onGet().reply(404, mockError);
  await store.dispatch(fetchPosts(fetchPostsActions));
  expect(store.getActions()).toMatchSnapshot();
});

it('should handle ' + FETCH_MORE_POSTS.success + ' action', async () => {
  mock.onGet().reply(200, postsResponse);
  await store.dispatch(fetchMorePosts(fetchMorePostsActions));
  expect(store.getActions()).toMatchSnapshot();
});

it('should handle ' + FETCH_MORE_POSTS.failure + ' action', async () => {
  const mockError = new Error('Look ma! I am an error.');
  mock.onGet().reply(404, mockError);
  await store.dispatch(fetchMorePosts(fetchMorePostsActions));
  expect(store.getActions()).toMatchSnapshot();
});

it('should handle ' + FETCH_REFRESHED_POSTS.success + ' action', async () => {
  mock.onGet().reply(200, postsResponse);
  await store.dispatch(fetchRefreshedPosts(fetchRefreshedPostsActions));
  expect(store.getActions()).toMatchSnapshot();
});

it('should handle ' + FETCH_REFRESHED_POSTS.failure + ' action', async () => {
  const mockError = new Error('Look ma! I am an error.');
  mock.onGet().reply(404, mockError);
  await store.dispatch(fetchRefreshedPosts(fetchRefreshedPostsActions));
  expect(store.getActions()).toMatchSnapshot();
});
