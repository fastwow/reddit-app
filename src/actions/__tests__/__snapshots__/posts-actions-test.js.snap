// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle fetch_more_posts_FAILURE action 1`] = `
Array [
  Object {
    "type": "fetch_more_posts_REQUEST",
  },
  Object {
    "error": "Ops! We have some internal error :)",
    "type": "fetch_more_posts_FAILURE",
  },
]
`;

exports[`should handle fetch_more_posts_SUCCESS action 1`] = `
Array [
  Object {
    "type": "fetch_more_posts_REQUEST",
  },
  Object {
    "data": undefined,
    "type": "fetch_more_posts_SUCCESS",
  },
]
`;

exports[`should handle fetch_posts_FAILURE action 1`] = `
Array [
  Object {
    "type": "fetch_posts_REQUEST",
  },
  Object {
    "error": "Ops! We have some internal error :)",
    "type": "fetch_posts_FAILURE",
  },
]
`;

exports[`should handle fetch_posts_SUCCESS action 1`] = `
Array [
  Object {
    "type": "fetch_posts_REQUEST",
  },
  Object {
    "data": undefined,
    "type": "fetch_posts_SUCCESS",
  },
]
`;

exports[`should handle fetch_refreshed_posts_FAILURE action 1`] = `
Array [
  Object {
    "type": "fetch_refreshed_posts_REQUEST",
  },
  Object {
    "error": "Ops! We have some internal error :)",
    "type": "fetch_refreshed_posts_FAILURE",
  },
]
`;

exports[`should handle fetch_refreshed_posts_SUCCESS action 1`] = `
Array [
  Object {
    "type": "fetch_refreshed_posts_REQUEST",
  },
  Object {
    "data": undefined,
    "type": "fetch_refreshed_posts_SUCCESS",
  },
]
`;
